src/commands/distribute/release.ts

```diff
@@ -0,0 +1,224 @@
+      }
+    } else {
+      return distributionGroupUsersRequestResponse.result;
+    }
+  }
+
+  private async createReleaseUpload(client: MobileCenterClient, app: DefaultApp): Promise<models.ReleaseUploadBeginResponse> {
+    const createReleaseUploadRequestResponse = await out.progress("Creating release upload...", 
+      clientRequest<models.ReleaseUploadBeginResponse>((cb) => client.distribute.createReleaseUpload(app.ownerName, app.appName, cb)));
+
+    if (createReleaseUploadRequestResponse.response.statusCode >= 400) {
+      throw failure(ErrorCodes.Exception, `failed to create release upload for ${this.filePath}`);
+    } else {
+      return createReleaseUploadRequestResponse.result;
+    }
+  }
+
+  private uploadFileToUri(uploadUrl: string, fileBuffer: Buffer, filename: string): Promise<void> {
+    debug("Uploading the release binary");
+    return new Promise<void>((resolve, reject) => {
+      Request.post({
+        formData: {
+          ipa: {
+            options: {
+              filename,
+              contentType: "application/octet-stream"
+            },
+            value: fileBuffer
+          }
+        },
+        url: uploadUrl
+      })
+      .on("error", (error) => {
+        reject(failure(ErrorCodes.Exception, `release binary uploading failed: ${error.message}`));
+      })
+      .on("response", (response) => {
+        if (response.statusCode < 400) {
+          resolve();
+        } else {
+          reject(failure(ErrorCodes.Exception, `release binary file uploading failed: HTTP ${response.statusCode} ${response.statusMessage}`));
+        }
+      });
+    });
+  }
+
+  private async finishReleaseUpload(client: MobileCenterClient, app: DefaultApp, uploadId: string): Promise<string> {
+    const finishReleaseUploadRequestResponse = await out.progress("Finishing release upload...", 
+      clientRequest<models.ReleaseUploadEndResponse>((cb) => client.distribute.updateReleaseUpload(uploadId, app.ownerName, app.appName, "committed", cb)));
+    
+    if (finishReleaseUploadRequestResponse.response.statusCode >= 400) {
+      throw failure(ErrorCodes.Exception, `failed to finish release upload for ${this.filePath}`);
+    } else {
+      return finishReleaseUploadRequestResponse.result.releaseUrl;
+    }
+  }
+
+  private async abortReleaseUpload(client: MobileCenterClient, app: DefaultApp, uploadId: string): Promise<void> {
+    const abortReleaseUploadRequestResponse = await out.progress("Aborting release upload...", 
+      clientRequest<models.ReleaseUploadEndResponse>((cb) => client.distribute.updateReleaseUpload(uploadId, app.ownerName, app.appName, "aborted", cb)));
+    
+    if (abortReleaseUploadRequestResponse.response.statusCode >= 400) {
+      throw new Error(`HTTP ${abortReleaseUploadRequestResponse.response.statusCode} - ${abortReleaseUploadRequestResponse.response.statusMessage}`);
+    }
+  }
+
+  private extractReleaseId(releaseUrl: string): number {
```

Comment:
Err.... yach. Now I understand the motivation for you question in the email. 
I didn't know the console methods doesn't allow passing the resource URL. 
I will might end up just adding the release id to the update upload response.